jobs/atc/spec

```diff
@@ -507,3 +507,42 @@ properties:
     description: |
       Client secret for CredHub authorization.
     default: ""
+
+  ssm.aws_region:
+    description: |
+      AWS region to use for fetching SSM parameters. Enviroment variable AWS_REGION is used
+      if this flag is not provided. This flag or the environment variable *must* be specified
+      for SSM secret backend to be enabled. When AWS_SDK_LOAD_CONFIG environment variable is
+      set, shared configuration files (~/.aws/config & ~/.aws/credentials) are used to load
+      this flag and credential flags. A profile within shared config file may be selected by
+      specifying it in AWS_PROFILE or AWS_DEFAULT_PROFILE environment variables.
+    default: ""
+  ssm.aws_access_key:
+    description: |
+      AWS Access key ID used as credentials for accessing SSM parameters. If this flag is not specified,
```

Comment:
Might as well not mention flags or env vars since this is the only way to set them.

Reply:
There are 4 ways in which AWS credentials can be provided:

1. Network interface: Available on EC2 instances and requires no configuration from user.
2. Credentials file in `~/.aws/`: Users with simple AWS setups will have these.
3. Environment variables
4. Command line flags

Out of these, only the last one requires some explicit code. All others are [provided by AWS SDK](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/sessions.html) and require no special implementation. I imagine people will require support for all of these methods (I have used all of them since I started working with AWS last year). I suggest either keeping the documentation or including a link to AWS SDK which explains various options.

Reply:
Sure, but with BOSH, no one should be doing anything but BOSH properties or EC2 instance metadata. The VMs are hands-off after deploying, and there should be no fussing with `~/.aws` or setting environment variables or anything.

I would not recommend documenting other methods since this is the BOSH release itself. I think it would only confuse people consuming the release.